primitives/bridge/src/snowbridge_outbound_token_transfer.rs

```diff
@@ -0,0 +1,557 @@
+    UniversalLocation,
+    EthereumNetwork,
+    OutboundQueue,
+    AgentHashedDescription,
+    ConvertAssetId,
+    BridgeChannelId,
+>(
+    PhantomData<(
+        UniversalLocation,
+        EthereumNetwork,
+        OutboundQueue,
+        AgentHashedDescription,
+        ConvertAssetId,
+        BridgeChannelId,
+    )>,
+);
+
+impl<
+        UniversalLocation,
+        EthereumNetwork,
+        OutboundQueue,
+        AgentHashedDescription,
+        ConvertAssetId,
+        BridgeChannelId,
+    > ExportXcm
+    for EthereumBlobExporter<
+        UniversalLocation,
+        EthereumNetwork,
+        OutboundQueue,
+        AgentHashedDescription,
+        ConvertAssetId,
+        BridgeChannelId,
+    >
+where
+    UniversalLocation: Get<InteriorLocation>,
+    EthereumNetwork: Get<NetworkId>,
+    OutboundQueue: SendMessage<Balance = u128>,
+    AgentHashedDescription: ConvertLocation<H256>,
+    ConvertAssetId: MaybeEquivalence<TokenId, Location>,
+    BridgeChannelId: Get<Option<ChannelId>>,
+{
+    type Ticket = (Vec<u8>, XcmHash);
+
+    fn validate(
+        network: NetworkId,
+        _channel: u32,
+        universal_source: &mut Option<InteriorLocation>,
+        destination: &mut Option<InteriorLocation>,
+        message: &mut Option<Xcm<()>>,
+    ) -> SendResult<Self::Ticket> {
+        let expected_network = EthereumNetwork::get();
+        let universal_location = UniversalLocation::get();
+
+        if network != expected_network {
+            log::trace!(target: "xcm::ethereum_blob_exporter", "skipped due to unmatched bridge network {network:?}.");
+            return Err(SendError::NotApplicable);
+        }
+
+        // Cloning destination to avoid modifying the value so subsequent exporters can use it.
+        let dest = destination
+            .clone()
+            .take()
+            .ok_or(SendError::MissingArgument)?;
+        if dest != Here {
+            log::trace!(target: "xcm::ethereum_blob_exporter", "skipped due to unmatched remote destination {dest:?}.");
+            return Err(SendError::NotApplicable);
+        }
+
+        // Cloning universal_source to avoid modifying the value so subsequent exporters can use it.
+        let (local_net, local_sub) = universal_source.clone()
+			.take()
+			.ok_or_else(|| {
+				log::error!(target: "xcm::ethereum_blob_exporter", "universal source not provided.");
+				SendError::MissingArgument
+			})?
+			.split_global()
+			.map_err(|()| {
+				log::error!(target: "xcm::ethereum_blob_exporter", "could not get global consensus from universal source '{universal_source:?}'.");
+				SendError::NotApplicable
+			})?;
+
+        if Ok(local_net) != universal_location.global_consensus() {
+            log::trace!(target: "xcm::ethereum_blob_exporter", "skipped due to unmatched relay network {local_net:?}.");
+            return Err(SendError::NotApplicable);
+        }
+
+        let source_location = Location::new(1, local_sub.clone());
```

Comment:
As we are sending directly from Tanssi, I believe we should use parents 0 here:

```suggestion
        let source_location = Location::new(0, local_sub.clone());
```

Reply:
Do you have more context on what `source_location` and `agent_id` obtained from it mean?

Reply:
I just checked again and parents 1 is correct, because it's the Ethereum chain the one which will interpret the message, so you can ignore my previous comment.

From what I understand, the `source_location` is the chain from which we are sending the message and the `agent_id` is the corresponding identifier of the snowbridge agent on Ethereum for that chain.